batchflow/tests/classification_metrics_test.py

```diff
@@ -0,0 +1,270 @@
+    (np.array([[0, 1], [1, 0]]), np.array([0, 1])),
+    (np.array([0, 1]), np.array([[0, 1], [1, 0]])),
+    (np.array([[[0, 1], [1, 0]], [[1, 1], [0, 0]]]), np.array([[0, 1], [1, 0]])),
+    (np.array([[0, 1], [1, 0]]), np.array([[[0, 1], [1, 0]], [[1, 1], [0, 0]]]))]
+
+TEST_PARAM_MULTICLASS = [
+    (np.array(2), np.array([[0.8, 0.1, 0.1], [0.1, 0.1, 0.8]])),
+    (np.array([[[0, 1, 0], [0, 1, 0], [1, 0, 0], [0, 0, 1]], [[1, 0, 0], [1, 1, 0], [1, 0, 1], [0, 0, 1]]]),
+     np.array([[0.1, 0.8, 0.1], [0.1, 0.8, 0.1], [0.8, 0.1, 0.1], [0.8, 0.1, 0.1]]))]
+
+# tests for confusion matrix
+
+def test_confusion_matrix():
+    """Test on correctness of confusion matrix calculation"""
+    y_true, y_pred = np.array([1, 1, 0, 1, 0, 0]), np.array([0, 0, 1, 0, 0, 0])
+    conf_matrix = np.array([[2, 3], [1, 0]])
+    conf_matrix_calc = cm(y_true, y_pred, fmt='labels', num_classes=2)._confusion_matrix  #pylint:disable=protected-access
+    assert (conf_matrix_calc == conf_matrix).all()
+
+def test_confusion_matrix_multiclass():
+    """Test on correctness of confusion matrix calculation for"""
+    y_true, y_pred = np.array([2, 1]), np.array([[0.1, 0.1, 0.8], [0.1, 0.8, 0.1]])
+    conf_matrix = np.array([[0, 0, 0], [0, 1, 0], [0, 0, 1]])
+    conf_matrix_calc = cm(y_true, y_pred, fmt='proba', axis=1, num_classes=3)._confusion_matrix  #pylint:disable=protected-access
+    assert (conf_matrix_calc == conf_matrix).all()
+
+# accuracy tests
+
+@pytest.mark.parametrize('y_true,y_pred', TEST_PARAMETERS)
+def test_diff_shapes_two_classes_acc(y_true, y_pred):
+    """Testing different shape"""
+    with pytest.raises(ValueError):
+        cm(y_true, y_pred, fmt='labels', num_classes=2).accuracy()
+
+@pytest.mark.parametrize('y_true,y_pred', TEST_PARAM_MULTICLASS)
+def test_diff_shapes_multiclass_acc(y_true, y_pred):
+    """Testing different shape"""
+    with pytest.raises(ValueError):
+        cm(y_true, y_pred, fmt='proba', axis=1, num_classes=3).accuracy()
+
+def test_single_value_two_class_acc():
+    """Test on accuracy single value in case of two class classification"""
+    y_true, y_pred = np.random.choice([0, 1], size=(5,)), np.random.choice([0, 1], size=(5,))
+    acc = cm(y_true, y_pred, fmt='labels', num_classes=2).accuracy()
+    assert isinstance(acc, np.floating)
+
+def test_vector_multiclass_acc():
+    """Test on accuracy multiclass"""
+    y_true = np.array([2, 1])
+    y_pred = np.array([[0.1, 0.1, 0.8], [0.1, 0.8, 0.1]])
+    acc = cm(y_true, y_pred, fmt='proba', axis=1, num_classes=3).accuracy()
+    assert isinstance(acc, np.floating)
+
+def test_vector_batches_two_class_acc():
+    """Test on accuracy vector with batch shape if input is a multidimensional array"""
+    y_true = np.array([[[1, 1], [0, 1]], [[0, 1], [1, 1]], [[1, 0], [1, 1]]])
+    y_pred = np.array([[[0, 1], [1, 1]], [[1, 0], [0, 0]], [[0, 0], [0, 1]]])
+    shape = cm(y_true, y_pred, fmt='labels', num_classes=2).accuracy().shape
+    assert shape == (3,)
```

Comment:
Since we want to check only shape of the result, we don't need thoroughly define `y_pred` and `y_true`.

If you want to use same values for multiple tests (e.g. for the same test for other metrics, or for different tests for the same metric), it could be a good idea to use `pytest.fixtures`

On the other hand, as was pointed out before, tests for different metrics could be united via `pytest.parametrize`, and that would also allow you to define `y_true` and `y_pred` fewer times